laravel/app/Models/Assignment.php

```diff
@@ -19,4 +19,29 @@ class Assignment extends Model
      * @var bool
      */
     public $timestamps = false;
+
+    public function residents()
+    {
+        return $this->belongsTo(Resident::class);
+    }
+
+    public function options()
+    {
+        return $this->belongsTo(Option::class);
+    }
+
+    public function scheduledatas()
+    {
+        return $this->belongsTo(ScheduleData::class);
+    }
+
+    public function attendings()
```

Comment:
should be singular `attending`, since an assignment only belongs to one Attending.